aspnetcore/blazor/file-uploads.md

```diff
@@ -92,6 +85,42 @@ await blobContainerClient.UploadBlobAsync(
 
 A component that receives an image file can call the <xref:Microsoft.AspNetCore.Components.Forms.BrowserFileExtensions.RequestImageFileAsync%2A?displayProperty=nameWithType> convenience method on the file to resize the image data within the browser's JavaScript runtime before the image is streamed into the app. Use cases for calling <xref:Microsoft.AspNetCore.Components.Forms.BrowserFileExtensions.RequestImageFileAsync%2A> are most appropriate for Blazor WebAssembly apps.
 
+## File size limit
+
+:::moniker range=">= aspnetcore-6.0"
+
+:::zone pivot="server"
+
+There's no file size limit for the <xref:Microsoft.AspNetCore.Components.Forms.InputFile> component in Blazor Server apps.
+
+:::zone-end
+
+:::zone pivot="webassembly"
+
+Prior to the release of ASP.NET Core 6.0, the <xref:Microsoft.AspNetCore.Components.Forms.InputFile> component had a file size limit of 2 GB. In ASP.NET Core 6.0 or later, the <xref:Microsoft.AspNetCore.Components.Forms.InputFile> component has no file size limit.
```

Comment:
[Moved from an issue comment by guardrex]

From https://learn.microsoft.com/en-us/aspnet/core/blazor/file-uploads?view=aspnetcore-5.0&pivots=webassembly:
![image](https://user-images.githubusercontent.com/22459191/216780767-59cdbed9-5b1d-4293-8069-6eb5297e1111.png)

is what you're referring to when you say:
> They fixed it at 6.0. Anyone who knows that it was limited in the past, perhaps from a PU issue that they might have come across while researching this, should have a clear statement that the limit isn't related to the component.

In response to my question of:
>  ...what are you even trying to accomplish with the first sentence?

I think there in lies the problem and confusion. That **Note** is a very vague statement. And we are unfortunately talking about two different things. That **Note** is simply referring to the ability to **read** the file contents as a stream. However, all of these issues are referring to the ability to **upload** a file. Granted, you can't accomplish the **upload** without the ability to **read**, but in my opinion, they are two, very different topics.

I may have a web application that **reads** a large (> 2GB) file completely on the client side, to gather statistics about a video file or something. That would be completely possible with C#, Blazor WASM, and the InputFile component now allowing file sizes larger than 2GB. But at no point in time am I trying to **upload** that file from the client to the server.

How about something like:
> There is no longer a file size limitation when using the \<xref:Microsoft.AspNetCore.Components.Forms.InputFile> component to read the contents of a file. However, there is a limit when attempting to upload a file using the \<xref:Microsoft.AspNetCore.Components.Forms.InputFile> component  in Blazor WebAssembly apps. When used for uploading, `BrowserHttpHandler.SendAsync` ([reference source](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs)) reads the file's bytes into a single JavaScript array buffer when marshalling the data from JavaScript to C#, so large file uploads (> 250 MB) may fail. For more information, see [Net6P7: Blazor WASM can't upload large files (500MB, 1GB, 2GB) (dotnet/aspnetcore #35899)](https://github.com/dotnet/aspnetcore/issues/35899).

Reply:
[I moved that last bit to a review comment so that we can resolve it. By the time Mackinnon and Pavel get here, it will be hidden away.]

Let me get back to on Monday morning. I have a ***HUGE*** move coming up that I'm packing for ... moving across the state of Florida 🚚😩. It's exciting, of course, moving to the 🚀 *Space Coast* region 👍 from the Pensacola area 👎, but the actual physical packing and moving part is 😩😩😩... *I'm getting too old for this sort of thing!* 😄 I'll get back to you on Monday, and we can iron that out before I ping Mackinnon and Pavel to assist further.

Reply:
Is so very unclear if you can do this or not. Is it just the limitation of InputFile or can it be done in some other way like here https://github.com/dotnet/aspnetcore/issues/43388#issuecomment-1222472290?

For those interested I´m trying to get more information [here ](https://github.com/Megabit/Blazorise/discussions/4557) in the Blazorise project.

Reply:
@sturlath ... Take that up on SO or chat about `XMLHttpRequest`. We won't cover that approach, even if it works, unless Mackinnon says later that it's ok to mention it. Currently, Javier has recommended streaming Fetch API or HTTP Ranges ... basically he's saying you handle the entire upload via JS. We probably won't be covering these approaches. We'll just be mentioning them, and devs should work within the community until the runtime folks update the runtime to address this on https://github.com/dotnet/runtime/issues/36634. I recommend that you 👍 that runtime issue. They need to know if devs want a feature, and lots of 👍 on an issue's opening comment is a good signal to them.